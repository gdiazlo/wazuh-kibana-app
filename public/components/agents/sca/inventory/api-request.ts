import { WzRequest } from '../../../../react-services/wz-request';

export async function getAgentPolicies(agentId) {
  const {
    data: {
      data: { affected_items: policies },
    },
  } = await WzRequest.apiReq('GET', `/sca/${agentId}`, {});

  return policies;
}
