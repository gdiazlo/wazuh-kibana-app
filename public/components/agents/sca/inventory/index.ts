export * from './api-request'
export * from './policies-table'