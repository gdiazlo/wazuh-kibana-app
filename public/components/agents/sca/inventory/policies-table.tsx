import { EuiBasicTable } from '@elastic/eui';
import React from 'react';

type Props = {
  policies: any[];
  columns: any[];
  rowProps?: any;
  loading?: boolean;
};

export default function SCAPoliciesTable(props: Props) {
  const { policies, columns, rowProps, loading = false } = props;

  const handleOnClick = (policy_id) => {
    rowProps(policy_id)
  }

  const getPoliciesRowProps = (item, idx) => {
    return {
      'data-test-subj': `sca-row-${idx}`,
      className: 'customRowClass',
      //This logic is necessary because when you click on a row in the table, it redirects you to a url that receives
      //a parameter and displays the desired table.
      onClick: rowProps ? () => handleOnClick(item.policy_id) : () => console.log('no row'),
    };
  };

  return (
    <>
      <EuiBasicTable items={policies} columns={columns} rowProps={getPoliciesRowProps} />
    </>
  );
}
