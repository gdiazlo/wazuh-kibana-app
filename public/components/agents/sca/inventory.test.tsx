import { fireEvent, render, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
// import {jest} from '@jest/globals'
import React from 'react';
import { Inventory } from './inventory'
import * as _ from 'lodash'
import { createGetterSetter } from '../../../utils/create-getter-setter';
import { ErrorOrchestratorService } from '../../../react-services/error-orchestrator/error-orchestrator.service';


describe('SCA Inventory Test', () => {
    // mocked getErrorOrchestrator
const mockedGetErrorOrchestrator = {
    handleError: jest.fn(),
  };

  jest.mock('../../../react-services/common-services', () => {
    return {
        getErrorOrchestrator: () => mockedGetErrorOrchestrator,
    };
  });
    // const mockGet = jest.fn();
    // const mockSet = jest.fn();
    // jest.mock('../../../react-services/error-orchestrator/error-orchestrator.service');
    // jest.mock('../../../utils/create-getter-setter');
    it('should render correctly', async () => {
        const spyReactConsoleError = jest.spyOn(console, 'error');
        spyReactConsoleError.mockImplementation(() => {});
        const onClickRow = jest.fn()
        const props= {
            agent: {
                agentPlatform:"linux",
                configSum:"ab73af41699f13fdd81903b5f23d8d00",
                dateAdd: "2022-09-01T15:40:06+00:00",
                group: ['default'],
                group_config_status: "synced",
                id: "001",
                ip: "172.18.0.5",
                lastKeepAlive: "2022-09-21T23:47:35+00:00",
                manager: "wazuh_manager_filebeat_sources_cmake-4.4-7.10.2",
                mergedSum: "4a8724b20dee0124ff9656783c490c4e",
                name: "wazuh_agent_ubuntu_sources_cmake-4.4",
                node_name: "manager-node",
                os: {arch: 'x86_64', codename: 'Bionic Beaver', major: '18', minor: '04', name: 'Ubuntu'},
                registerIP: "any",
                status: "active",
                version: "Wazuh v4.4.0",
            },
            onClickRow,
            withoutDashboard: true
        } 
        // try {

            const { asFragment, debug } = render(<Inventory {...props} />);
        debug()

            await waitFor(() => expect(asFragment()).toMatchSnapshot());
            // await waitFor(() => expect(createGetterSetter).toHaveBeenCalledTimes(1))
            // await waitFor(() => expect(ErrorOrchestratorService).toHaveBeenCalledTimes(1))
        // } catch(error) {
        //     console.log(error)
        // }
        // expect(asFragment()).toMatchSnapshot();
        // const { debug } = render(<Inventory {...props} />);  

})
});
