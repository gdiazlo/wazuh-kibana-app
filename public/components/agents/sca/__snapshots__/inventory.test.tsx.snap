// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SCA Inventory Test should render correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      style="margin: 16px;"
    >
      <div
        class="euiSpacer euiSpacer--m"
      />
      <div
        class="euiProgress euiProgress--indeterminate euiProgress--xs euiProgress--primary"
      />
    </div>
  </div>
  <div
    class="euiPage"
  />
</DocumentFragment>
`;

exports[`SCA Inventory Test should render correctly 2`] = `
<DocumentFragment>
  <div />
  <div
    class="euiPage"
  >
    <div
      class="euiCallOut euiCallOut--primary"
    >
      <div
        class="euiCallOutHeader"
      >
        <svg
          aria-hidden="true"
          class="euiIcon euiIcon--medium euiCallOutHeader__icon"
          focusable="false"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.5 11.508L7.468 8H6.25V7h2.401l.03 3.508H9.8v1H7.5zm-.25-6.202a.83.83 0 01.207-.577c.137-.153.334-.229.59-.229.256 0 .454.076.594.23.14.152.209.345.209.576 0 .228-.07.417-.21.568-.14.15-.337.226-.593.226-.256 0-.453-.075-.59-.226a.81.81 0 01-.207-.568zM8 13A5 5 0 108 3a5 5 0 000 10zm0 1A6 6 0 118 2a6 6 0 010 12z"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="euiCallOutHeader__title"
        >
          No scans available
        </span>
      </div>
      <div
        class="euiText euiText--small"
      >
        <button
          class="euiButton euiButton--primary"
          type="button"
        >
          <span
            class="euiButtonContent euiButton__content"
          >
            <span
              class="euiButton__text"
            >
              Refresh
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SCA Inventory Test should render correctly 3`] = `<DocumentFragment />`;
